rfcs/convergence/deprecate-keyboard-key-package.md

```diff
@@ -0,0 +1,82 @@
+# RFC: Handle keyboard keys in more modern and tree shakeable way
+
+@ling1726
+
+## Summary
+
+Proposes to break the dependency of any v9 packages on `@fluentui/keyboard-key` since it is not a converged package and can increase bundle size.
+Also proposes to stop using `keyCode` in v9 components and start using `key`.
+
+## Problem statement
+
+```typescript
+import { ArrowLeftKey } from '@fluentui/keyboard-key';
+
+console.log(ArrowLeftKey);
+```
+
+The above code will increase the bundle size by **2kb** minified and **1.0kb** gzipped. Looking at the output, every single key is retained in the bundle. The measurement was done using the bunde-size tool implemented as a part of [#17697](https://github.com/microsoft/fluentui/issues/17697).
+
+According to [MDN](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode) `keyCode` has been deprecated from some time and developers should prefer to use `e.key` which takes into consideration modifier keys and the locale and layout. IE11 handles the values differently to other evergreen browsers, a few examples:
+
+| Everyone else | IE11     |
+| ------------- | -------- |
+| ArrowDown     | Down     |
+| Escape        | Esc      |
+| ' '           | Spacebar |
+
+## Detailed Design or Proposal
+
+### Remove dual v8/v9 dependency on `@fluentui/keyboard-key`
+
+Since there are v9 components now being used in HVCs and products, this might be difficult since this could result in a breaking change in terms of dependencies.
+
+#### Option 1: Keep using keyboard-key
+
+We can copy the contents of keyboard-key internally into the v8 `react` folder. We have already done this previously with v7. This lets us modify keyboard-key freely without breaking v8.
```

Comment:
v8 wasn't using keyboard-key at all until a previous convergence attempt, so I think we really ought to just get rid of it from v8 (and go back to whatever the previous implementation was)

Reply:
However previous versions of v8/7 will use this and some partners (Teams) have a hard resolution for a specific version of `keyboard-key` this makes any changes we make there hard for customers to consume

Reply:
however if you mean remove the package from the repo completely ... sounds good to me

Reply:
What I mean is:
- Remove the keyboard-key dep from v8 and revert to old implementation. I can't see this causing any problems for partners, even if they have specific resolutions for keyboard-key.
- TBD for v0, whatever seems appropriate to you
- Possibly remove the package from the repo, if nothing in the repo is using the current version and we don't expect to need to release fixes for it